lib/ultrajsondec.c

```diff
@@ -131,15 +131,17 @@ static FASTCALL_ATTR JSOBJ FASTCALL_MSVC decode_numeric (struct DecoderState *ds
       case '8':
       case '9':
       {
+        // detect overflow on digit shift
+        if ((intValue * 10ULL) % 10 != 0)
+        {
+          hasError = 1;
+        }
+
         //PERF: Don't do 64-bit arithmetic here unless we know we have to
         prevIntValue = intValue;
         intValue = intValue * 10ULL + (JSLONG) (chr - 48);
 
-        if (intNeg == 1 && prevIntValue > intValue)
-        {
-          hasError = 1;
-        }
-        else if (intNeg == -1 && intValue > overflowLimit)
```

Comment:
Doesn't the removal of this check mean that there is no test against the smaller `overflowLimit` of `LLONG_MAX` for negative numbers anymore? Then we could get an overflow further down in the `BREAK_INT_LOOP` section.